src/map/fov.cpp

```diff
@@ -0,0 +1,475 @@
+		}
+		PrepareShadowCaster(player, unit, pos, marker);
+		PrepareCache(pos, width, height, range);
+		ProceedShadowCasting(pos, width, height, range + 1);
+		ResetShadowCaster();
+	} else {
+		ProceedSimpleRadial(player, pos, width, height, range, marker);
+	}
+}
+
+/**
+**  Refresh the whole sight of unit by SimleRadial algorithm. (Explore and make visible.)
+**
+**  @param player  player to mark the sight for (not unit owner)
+**  @param pos     location to mark
+**  @param w       width to mark, in square
+**  @param h       height to mark, in square
+**  @param range   Radius to mark (sight range)
+**  @param marker  Function to mark or unmark sight
+*/
+void CFieldOfView::ProceedSimpleRadial(const CPlayer &player, const Vec2i &pos, 
+									   const int16_t w, const int16_t h, const int16_t range, 
+									   MapMarkerFunc *marker) const
+{
+	// Up hemi-cyle
+	const int16_t miny = std::max(-range, 0 - pos.y);
+	for (int16_t offsety = miny; offsety != 0; offsety++) {
+		const int16_t offsetx = isqrt(square(range + 1) - square(-offsety) - 1);
+		const int16_t minx = std::max(0, pos.x - offsetx);
+		const int16_t maxx = std::min(Map.Info.MapWidth, pos.x + w + offsetx);
+		Vec2i mpos(minx, pos.y + offsety);
+		const size_t index = mpos.y * Map.Info.MapWidth;
+
+		for (mpos.x = minx; mpos.x < maxx; mpos.x++) {
+			marker(player, mpos.x + index);
+		}
+	}
+	for (int16_t offsety = 0; offsety < h; offsety++) {
+		const int16_t minx = std::max(0, pos.x - range);
+		const int16_t maxx = std::min(Map.Info.MapWidth, pos.x + w + range);
+		Vec2i mpos(minx, pos.y + offsety);
+		const size_t index = mpos.y * Map.Info.MapWidth;
+
+		for (mpos.x = minx; mpos.x < maxx; mpos.x++) {
+			marker(player, mpos.x + index);
+		}
+	}
+	// bottom hemi-cycle
+	const int16_t maxy = std::min(range, int16_t(Map.Info.MapHeight - pos.y - h));
+	for (int16_t offsety = 0; offsety < maxy; offsety++) {
+		const int16_t offsetx = isqrt(square(range + 1) - square(offsety + 1) - 1);
+		const int16_t minx = std::max(0, pos.x - offsetx);
+		const int16_t maxx = std::min(Map.Info.MapWidth, pos.x + w + offsetx);
+		Vec2i mpos(minx, pos.y + h + offsety);
+		const size_t index = mpos.y * Map.Info.MapWidth;
+
+		for (mpos.x = minx; mpos.x < maxx; mpos.x++) {
+			marker(player, mpos.x + index);
+		}
+	}
+}
+
+/**
+** Mark the sight of unit by ShadowCaster algorithm. (Explore and make visible.)
+**
+**  @param spectratorPos	Tile position of the spectrator unit - upper left corner for unit larger than 1x1
+**  @param width			Spectrator's width in tiles
+**  @param height			Spectrator's height in tiles
+**  @param range			Spectrator's sight range in tiles
+*/
+void CFieldOfView::ProceedShadowCasting(const Vec2i &spectatorPos, const uint16_t width, const uint16_t height, const uint16_t range)
+{
+	enum SpectatorGeometry {cOneTiled, cEven, cOdd, cTall, cWide} ;
+	const uint8_t geometry = [width, height]{   if (width == height) {
```

Comment:
Because it's an initialization of `const`

Reply:
I was just wondering about the performance of this, since this is called often. Maybe it would be better (even if it's more ugly) to have the code inline

Reply:
Hmmm, do you think this lambda has such dramatic overhead? Or do you mean to inline the whole `ProceedShadowCasting()`?